lib/celluloid/io/ssl_socket.rb

```diff
@@ -7,12 +7,13 @@ class SSLSocket < Stream
       extend Forwardable
 
       def_delegators :@socket, :read_nonblock, :write_nonblock, :close, :closed?,
-        :cert, :cipher, :client_ca, :peer_cert, :peer_cert_chain, :verify_result, :peeraddr
+        :cert, :cipher, :client_ca, :peer_cert, :peer_cert_chain, :verify_result, :peeraddr, :sync_close=
 
       def initialize(io, ctx = OpenSSL::SSL::SSLContext.new)
         super()
         @context = ctx
         @socket = OpenSSL::SSL::SSLSocket.new(::IO.try_convert(io), @context)
+        @socket.sync_close = true if @socket.respond_to?(:sync_close=)
```

Comment:
No problem! I agree on it being ugly. I hate the sync_close setter but Ruby made it default off so :(

Sent from my iPhone

> On Apr 6, 2015, at 11:43, digitalextremist <notifications@github.com> wrote:
> 
> In lib/celluloid/io/ssl_socket.rb:
> 
> >  
> >        def initialize(io, ctx = OpenSSL::SSL::SSLContext.new)
> >          super()
> >          @context = ctx
> >          @socket = OpenSSL::SSL::SSLSocket.new(::IO.try_convert(io), @context)
> > +        @socket.sync_close = true if @socket.respond_to?(:sync_close=)
> My bad on confusing the intent here, focusing only on function. Clear to merge then once the blizzard of tests is subdued to something sane.
> 
> —
> Reply to this email directly or view it on GitHub.
>